dds/idl/marshal_generator.cpp

```diff
@@ -122,6 +122,51 @@ namespace {
     },
   };
 
+  string type_to_default(AST_Type* type, const string& name) {
+    string val;
+    AST_Type* actual_type = resolveActualType(type);
+    Classification fld_cls = classify(actual_type);
+    const bool use_cxx11 = be_global->language_mapping() == BE_GlobalData::LANGMAP_CXX11;
+    if ((fld_cls & CL_STRUCTURE) || (fld_cls & CL_UNION)) {
+      val = string("set_default(") + name + ");\n";
+   // default is an array of the same dimensions and element type whose elements take the default value for their corresponding type
+
+   // a union with the discriminatorset to the default value for the discriminator type.
+    //if this selects a branch then the selected member is also set to the default value for the member type.
+    //otherwise the value of the union is fully specified by the discriminator value
+
+    } else if (fld_cls & CL_ARRAY) {
+      // TODO
+      //val = string("set_default(") + name + ");\n";
+    } else if (fld_cls & CL_ENUM) {
+      // For now, simply return the first value of the enumeration.
+      // Must be changed, if support for @default_literal is desired.
+      AST_Enum* enu = AST_Enum::narrow_from_decl(actual_type);
```

Comment:
```suggestion
      AST_Enum* enu = dynamic_cast<AST_Enum*>(actual_type);
```

This goes for any `narrow_from_decl`.